pkg/util/util.go

```diff
@@ -149,6 +152,32 @@ func IsNodeReady(node *v1.Node) bool {
 	return false
 }
 
+// MachineToInfrastructureMapFunc returns a handler.ToRequestsFunc that watches for
+// Machine events and returns reconciliation requests for an infrastructure provider object.
+func MachineToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToRequestsFunc {
```

Comment:
Why in `util` and not in the machine controller code?

Reply:
It feels weird to import the machine controller code from a provider controller, wdyt?

Reply:
Oh that's right, this is meant to be used by infra providers. It'd be kinda nice to see this in something like `pkg/machine` as `InfrastructureMapFunc` - i.e. `machine.InfrastructureMapFunc`. But I realize having a package named `machine` would conflict with having a `Machine` with the same name.

Long-term, I want to eliminate `pkg/util` and move everything into other packages. `util` is too much of a grab bag.

I suppose it's ok to lump it in here for now 😄